File: openapi-generator/dart-jaguar/lib/model/vehicle.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

part 'vehicle.jser.dart';

class Vehicle {
   /* Total engine hours for the vehicle. */
  @Alias('engineHours')
  final int engineHours;
   /* The fuel level of the vehicle as a percentage. (0.0 to 1.0) */
  @Alias('fuelLevelPercent')
  final double fuelLevelPercent;
   /* ID of the vehicle. */
  @Alias('id')
  final int id;
   /* Name of the vehicle. */
  @Alias('name')
  final String name;
  
  @Alias('note')
  final String note;
   /* The number of meters reported by the odometer. */
  @Alias('odometerMeters')
  final int odometerMeters;
   /* Vehicle Identification Number. */
  @Alias('vin')
  final String vin;
  

  Vehicle(
      

{
     this.engineHours = null,  
     this.fuelLevelPercent = null,  
    
     this.id = null,   this.name = null,  
     this.note = null,  
     this.odometerMeters = null,  
     this.vin = null 
    
    }
  );

  @override
  String toString() {
    return 'Vehicle[engineHours=$engineHours, fuelLevelPercent=$fuelLevelPercent, id=$id, name=$name, note=$note, odometerMeters=$odometerMeters, vin=$vin, ]';
  }
}

@GenSerializer(nullableFields: true)
class VehicleSerializer extends Serializer<Vehicle> with _$VehicleSerializer {

}

File: openapi-generator/php-ze-ph/src/App/DTO/Vehicle.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * A vehicle object.
 */
class Vehicle
{
    /**
     * Total engine hours for the vehicle.
     * @DTA\Data(field="engineHours", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"int"})
     * @var int
     */
    public $engine_hours;
    /**
     * The fuel level of the vehicle as a percentage. (0.0 to 1.0)
     * @DTA\Data(field="fuelLevelPercent", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"float"})
     * @var float
     */
    public $fuel_level_percent;
    /**
     * ID of the vehicle.
     * @DTA\Data(field="id")
     * @DTA\Validator(name="Type", options={"type":"int"})
     * @var int
     */
    public $id;
    /**
     * Name of the vehicle.
     * @DTA\Data(field="name", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $name;
    /**
     * @DTA\Data(field="note", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $note;
    /**
     * The number of meters reported by the odometer.
     * @DTA\Data(field="odometerMeters", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"int"})
     * @var int
     */
    public $odometer_meters;
    /**
     * Vehicle Identification Number.
     * @DTA\Data(field="vin", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $vin;
}
